src/Xamarin.Android.Build.Tasks/Utilities/MonoAndroidHelper.cs

```diff
@@ -830,5 +830,22 @@ public static AndroidRuntime ParseAndroidRuntime (string androidRuntime)
 			// Default runtime is MonoVM
 			return AndroidRuntime.MonoVM;
 		}
+
+		/// <summary>
+		/// Returns true if the input file is newer than the output file.
+		/// </summary>
+		public static bool IsFileOutOfDate (string input, string output)
```

Comment:
Is this logic in a separate target? We could use MSBuild's Inputs and Outputs to do this instead?

* https://github.com/dotnet/android/blob/main/Documentation/guides/MSBuildBestPractices.md#building-partially

And then could test it will this helper method:

https://github.com/dotnet/android/blob/ad6cc1b4b8b0075ccc942846028326683bc5400e/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/AndroidUpdateResourcesTest.cs#L237

Reply:
I initially did it this way, but I didn't like that it calls the target X00 individual times.  This flooded the binlog making it a lot harder to read.

From reading the docs, I had thought it would call the target once, providing an ItemGroup of inputs that needed to be updated, but that was not the case.

Reply:
Interesting, `_LinkAssembliesNoShrink` does appear to do exactly what I wanted to do:

![image](https://github.com/user-attachments/assets/251678e0-8876-4e54-932f-b82e90f77c61)

Here's what I was using and I don't understand why it works differently:

```xml
<Target Name="_CompressAssemblies"
    DependsOnTargets="_CollectAssembliesToCompress"
    Inputs="@(_AssembliesToCompress)"
    Outputs="@(_AssembliesToCompress->%(DestinationPath))"
    Condition="'$(EmbedAssembliesIntoApk)' == 'True'">

    <CompressAssembly
        AssembliesToCompress="@(_AssembliesToCompress)">
      <Output TaskParameter="CompressionFailedAssemblies" ItemName="_CompressionFailedAssemblies" />
    </CompressAssembly>
</Target>
```

Reply:
I would normally put single quote around: `Outputs="@(_AssembliesToCompress->'%(DestinationPath)')"`

Other than that, maybe it's the `<Output/>` that makes it behave differently?

Reply:
Huh, I guess the single quote was important.  Without it I get:

![image](https://github.com/user-attachments/assets/dc1f03fb-8337-41bb-9b0a-ad126676a126)

Adding it makes it work as expected:

![image](https://github.com/user-attachments/assets/9d92af43-a9d0-4c91-a309-39bc27ea7a3c)

Thanks for the help!